components/board/genericlinux/board.go

```diff
@@ -492,6 +492,18 @@ func (b *Board) WriteAnalog(ctx context.Context, pin string, value int32, extra
 	return nil
 }
 
+// StreamTicks starts a stream of digital interrupt ticks.
+func (b *Board) StreamTicks(ctx context.Context, interrupts []string, ch chan board.Tick, extra map[string]interface{}) error {
+	for _, name := range interrupts {
+		interrupt, ok := b.DigitalInterruptByName(name)
+		if !ok {
+			return errors.Errorf("unknown digital interrupt: %s", name)
+		}
+		interrupt.AddCallback(ch)
```

Comment:
I see us adding these callbacks. Can we ever remove them, for example when the thing on the other side disconnects?

Reply:
:exploding_head: Dang, we _need_ to be able to remove the old ones! If you stream ticks, disconnect, and try streaming ticks again without removing the callback, the second channel will never get anywhere because we'll block waiting to send ticks to the first channel!